components/kyma-environment-broker/internal/process/update/initialisation_step.go

```diff
@@ -64,6 +64,10 @@ func (s *InitialisationStep) Run(operation internal.UpdatingOperation, log logru
 		if err != nil {
 			return operation, time.Second, nil
 		}
+		// because below lines would overwrite the trigger taking the instance details from
+		// previous operation while the update is in state Pending, this ensures that we
+		// overwrite the overwritten overwrites...
+		instance.InstanceDetails.SCMigrationTriggered = operation.InstanceDetails.SCMigrationTriggered
```

Comment:
@ksputo @piotrmiskiewicz why do we overwrite instance details in the update operation?
because I think this prevents any changes to instance details in the PATCH instance endpoint. What I observed to happen is:
* I set the migration flag in the instance details
* the update operation gets created with correct instance details
* the update/iinitialisation_step looks up the instance details from DB
* this means taking the lastOperation, that didn't have the migration flag obviously because it was an entirely different operation
https://github.com/kyma-project/control-plane/blob/5bb0ad9c1d451d83313964ba3cb76f9aa6817407/components/kyma-environment-broker/internal/storage/driver/postsql/instance.go#L326-L333
* overwrite instance details with migration flag for the update operation to always false unless it was present for provisioning request already (which wouldn't make a lot of sense imo)

Reply:
populate instance details migration triggered from `operation.provisioningparameters.erscontext` instead